shell.nix

```diff
@@ -0,0 +1,15 @@
+# just qick way to get exact rust used with all components and targets on mac and linux
```

Comment:
I'd like to avoid commiting tool-specific configs like this into the repo. I think most of the developers just use `rustup` to install rust, and I don't want to make it more complex than that in this repo, given that it's just a rust crate that doesn't depend on any OS-level libraries and binaries.